### PR TITLE
auth policy: Document fail_type

### PR DESCRIPTION
**Pull request by cmouse**
JIRA: DOV-5496

### CODE REVIEW

File: source/configuration_manual/authentication/auth_policy.rst [hunk 1]
```diff
@@ -168,6 +168,46 @@ produces
 
 for this to work.
 
+List of fields
+--------------
+
+All fields supported by :ref:variables-auth can be used. In addition, you can use following fields:
+
+``hashed_password``
+  User's password hashed with password hash algorithm.
```
Comment (sirainen) [inline source/configuration_manual/authentication/auth_policy.rst:177]: I guess based on auth_policy_hash_* settings? Could have some pointer there. Or "with the configured password hash algorithm."

File: source/configuration_manual/authentication/auth_policy.rst [hunk 2]
```diff
@@ -168,6 +168,46 @@ produces
...
+
+``requested_username``
+  Username that was requested. This is normalization field which contains the correct username,
```
Comment (sirainen) [inline source/configuration_manual/authentication/auth_policy.rst:180]: What does "normalization field" mean here?

File: source/configuration_manual/authentication/auth_policy.rst [hunk 3]
```diff
@@ -168,6 +168,46 @@ produces
...
+  depending on wheter this is master user login or not.
+
+``fail_type``
+  .. versionadded:: v3.0.0;v2.4.0
+
+  Will be filled with the reason request failed. This can take one of
```
Comment (sirainen) [inline source/configuration_manual/authentication/auth_policy.rst:186]: * "Will be filled with" seems unnecessary. Just start with "The reason .."
 * the request

File: source/configuration_manual/authentication/auth_policy.rst [hunk 4]
```diff
@@ -168,6 +168,46 @@ produces
...
+  ``credentials``
+     The user's credentials were wrong.
+  ``account``
+     Account was invalid or disabled.
+  ``expired``
+     User's password is expired.
+  ``policy``
+     Login was rejected by policy server.
+
+Following fields are always present.
```
Comment (sirainen) [inline source/configuration_manual/authentication/auth_policy.rst:199]: * The following
 * present: with colon

**Review by sirainen — CHANGES_REQUESTED** [review]

File: source/configuration_manual/authentication/auth_policy.rst (hunk 1 above)
Comment (cmouse) [reply source/configuration_manual/authentication/auth_policy.rst:177]: it literally says "password hash algorithm" which is title few lines up =)

but i'll put a link to it.

File: source/configuration_manual/authentication/auth_policy.rst [hunk 5]
```diff
@@ -168,6 +170,48 @@ produces
...
+--------------
+
+All fields supported by :ref:`variables-auth` can be used. In addition, you can use following fields:
+
+``hashed_password``
+  User's password hashed with :ref:`authentication-policy_password_hash_algorithm`.
+
+``requested_username``
+  Username that was requested. This is a field which contains the correct username,
+  depending on whether this is master user login or not.
```
Comment (sirainen) [inline source/configuration_manual/authentication/auth_policy.rst:183]: Still a bit unclear what is "the correct username". I guess the intention is something like: `Username for regular logins. For master user logins, this is the requested login username (not the master username).`?

**Review by sirainen — CHANGES_REQUESTED** [review]

**Review by sirainen — APPROVED** [review]

**Pull request merged**